app/src/main/java/com/example/songme/ui/main/MainActivity.kt

```diff
@@ -2,42 +2,116 @@ package com.example.songme.ui.main
+import android.content.ServiceConnection
 import android.content.pm.PackageManager
 import android.os.Build
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.os.Handler
+import android.os.IBinder
 import android.view.MenuItem
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
 import androidx.core.app.ActivityCompat
 import androidx.fragment.app.Fragment
+import com.bumptech.glide.Glide
 import com.example.songme.R
+import com.example.songme.data.model.Track
+import com.example.songme.service.ServiceContract
+import com.example.songme.service.MusicService
 import com.example.songme.ui.genres.GenresFragment
 import com.example.songme.ui.home.ActionBarFragment
 import com.example.songme.ui.home.HomeFragment
 import com.example.songme.ui.mymusic.MyMusicFragment
+import com.example.songme.ui.adapter.TrackAdapter.OnSendDataSelectedListener
 import com.example.songme.utils.Constants.STORAGE_PERMISSION_CODE
+import com.example.songme.utils.Constants.TIME_FORMAT
+import com.example.songme.utils.Constants.TIME_UPDATE_TRACK
 import com.example.songme.utils.showMessage
 import com.google.android.material.bottomnavigation.BottomNavigationView
+import com.google.android.material.bottomsheet.BottomSheetBehavior
 import kotlinx.android.synthetic.main.content_main.*
+import kotlinx.android.synthetic.main.play_music_screen.*
+import java.text.SimpleDateFormat
+
+class MainActivity : AppCompatActivity(),
+    BottomNavigationView.OnNavigationItemSelectedListener,
+    OnSendDataSelectedListener,
+    ServiceContract.OnMediaPlayChange,
+    View.OnClickListener {
+
+    private var musicService: MusicService? = null
+    private var isBound = false
+    private var sheetBehavior: BottomSheetBehavior<ConstraintLayout>? = null
+    private var handler: Handler? = null
+    private val serviceConnection: ServiceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            val binder = service as MusicService.TrackBinder
+            musicService = binder.getService()
+            musicService?.setOnMediaChangeListener(this@MainActivity)
+            updateSeekBar()
+            isBound = true
+        }
+
+        override fun onServiceDisconnected(name: ComponentName?) {
+            isBound = false
+        }
+    }
 
-class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         initComponents()
         initListeners()
     }
 
+    override fun onStart() {
+        super.onStart()
+        if (!isBound) {
+            val intent = Intent(applicationContext, MusicService::class.java)
+            bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE)
```

Comment:
sao lại khởi tạo Intent ở đây 
```suggestion
            bindService(MusicService.getIntent(), serviceConnection, Context.BIND_AUTO_CREATE)
```